core/src/main/scala/kafka/log/Log.scala

```diff
@@ -527,8 +527,10 @@ class Log(val dir: File,
     if(numToDelete > 0) {
       lock synchronized {
         // we must always have at least one segment, so if we are going to delete all the segments, create a new one first
-        if(segments.size == numToDelete)
+        if(segments.size == numToDelete) {
           roll()
+          updateLogEndOffset(nextOffsetMetadata.messageOffset)
```

Comment:
Maybe we can put this inside the roll()? Right after we add the new segment into the segments list. Because that is the actual place the base offset changes.